gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -2196,6 +2197,17 @@ public default GraphTraversal<S, String> format(final String format) {
         return this.asAdmin().addStep(new FormatStep<>(this.asAdmin(), format));
     }
 
+    /**
+     * Parse the incoming traverser as a boolean value.
+     *
+     * @return the traversal with an appended {@link AsBoolStep}.
+     * @see <a href="http://tinkerpop.apache.org/docs/${project.version}/reference/#asBool-step" target="_blank">Reference Documentation - asBool Step</a>
+     */
+    public default GraphTraversal<S, Boolean> asBool() {
+        this.asAdmin().getGremlinLang().addStep(Symbols.asBool);
```

Comment:
I see this work is based off of the master branch, we would want it to instead be based off of `3.8-dev`. The difference is that `getGremlinLang()` will need to be replaced with `getBytecode()` in 3.8